### DISCUSSION TITLE
ScriptReference/SceneManagement.EditorSceneManager.OpenScene

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#145] [Notes]
# ScriptReference/SceneManagement.EditorSceneManager.OpenScene

https://docs.unity3d.com/ScriptReference/SceneManagement.EditorSceneManager.OpenScene.html

**Comment by vertxxyz** [comment]
This will trigger a [`Resources.UnloadUnusedAssets`](https://docs.unity3d.com/ScriptReference/Resources.UnloadUnusedAssets.html), which can cause Objects that are only referenced on the stack to be unloaded.  
If you're getting a MissingReferenceException after opening a scene, consider storing the reference in an editor window or static variable to remain loaded through the transition.

**Comment by vertxxyz** [comment]
You can use [`EditorBuildSettings.scenes`](https://docs.unity3d.com/ScriptReference/EditorBuildSettings-scenes.html) to get the path of a scene from its build index.

**Comment by electronics45** [comment]
Note that this method loads the scene synchronously, so the returned scene will have `scene.isLoaded` return true, and you can immediately perform operations on it's GameObjects/etc, without having to wait for a `SceneManager.sceneLoaded` event.
